File: Data Structures & Algorithms/is-palindrome/submission-2.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int left = 0;
        int right = s.Length - 1;

        while (left < right)
        {
            for (var i = left;i < right;i++)
            {
                if (!char.IsLetterOrDigit(s[i]))
                    left++;
                else
                    break;
            }

            for (var i = right;i > left;i--)
            {
                if (!char.IsLetterOrDigit(s[i]))
                    right--;
                else
                    break;
            }

            if (char.ToLower(s[left]) != char.ToLower(s[right]))
                return false;
            left++;
            right--;
        }
        return true;        
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left, right = 0, len(nums) - 1
        minIndex = -1

        while left <= right:
            mid = left + (right - left)//2
            if nums[mid] <= nums[-1]:
                right = mid - 1
                minIndex = mid
            else:
                left = mid + 1
        
        return nums[minIndex]        